core/pkg/server/sender.go

```diff
@@ -561,7 +561,11 @@ func (s *Sender) sendRequestDefer(request *service.DeferRequest) {
 		s.fwdRequestDefer(request)
 	case service.DeferRequest_FLUSH_FS:
 		if s.fileStream != nil {
-			s.fileStream.Close()
+			if s.exitRecord != nil {
+				s.fileStream.FinishWithExit(s.exitRecord.GetExit().GetExitCode())
+			} else {
+				s.fileStream.FinishWithoutExit()
```

Comment:
same comment as above about getting into this state. i think there some subtlety here with the last message with complete and exit code, i'm worried that we will have runs that look like these are running for a long time.

I don't see how we won't have exit code, unless something bad happened

Reply:
Added `CaptureError` for this case!